agent/grpc-external/server.go

```diff
@@ -71,6 +101,11 @@ func NewServer(
 		}),
 	}
 
+	// the caller wants to forward the traffic to the consul server if presents the serverConn
+	if serverConn != nil {
+		opts = append(opts, grpc.UnknownServiceHandler(proxy.TransparentHandler(makeDirector(serverConn))))
```

Comment:
Should this really be using an `UnknownServiceHandler`? From what I recall, there are only a select few resource endpoints, so perhaps we should explicitly define them, rather than setting a catch-all handler like this.

Reply:
addressed in another thread